stdlib/pathlib.pyi

```diff
@@ -39,6 +46,8 @@ class PurePath(PathLike[str]):
     def suffixes(self) -> list[str]: ...
     @property
     def stem(self) -> str: ...
+    if sys.version_info >= (3, 13):
+        def full_match(self, pattern: str, *, case_sensitive: bool | None = None) -> bool: ...
```

Comment:
This method is actually implemented on a new base class that has been introduced in Python 3.13, `PurePathBase`. For the `pattern` argument, it also accepts instances of `PurePathBase` as well as `str`s: https://github.com/python/cpython/blob/2268289a47c6e3c9a220b53697f9480ec390466f/Lib/pathlib/_abc.py#L395-L396

Reply:
Can `PurePathBase` be accessed at runtime? `PurePath` is inheriting from `PurePathBase` so we could maybe use  `def full_match(self, pattern: str | PurePath, *, case_sensitive: bool | None = None) -> bool`?

Reply:
`PurePath` can be constructed using arbitrary `os.PathLike` objects, so I changed the signature for `PurePath.full_match` in https://github.com/python/typeshed/pull/12048/commits/89b7469898d44da8f0e348e05cb8120d77fd149b to accept `_typeshed.StrPath` for `pattern` parameter. `_typeshed.StrPath` is a union of `str | os.PathLike[str]`.

`PurePathBase` _can_ be accessed at runtime, but only if you import the private `pathlib._abc` module. I don't think we should worry about that in this PR :-)